test/unit/util.test.ts

```diff
@@ -412,4 +413,49 @@ describe('Utils', () => {
       await expect(extractArticleList('http://valid-wrong-url.com/')).rejects.toThrow('Failed to read articleList from URL: http://valid-wrong-url.com/')
     })
   })
+
+  describe('metaData', () => {
```

Comment:
Your automated tests don't really test the edge cases with undefined, wrong type, missing values.